modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/layout/display/page/FileEntryLayoutDisplayPageProvider.java

```diff
@@ -70,16 +74,33 @@ public String getClassName() {
 	public LayoutDisplayPageObjectProvider<FileEntry>
 		getLayoutDisplayPageObjectProvider(long groupId, String urlTitle) {
 
-		return getLayoutDisplayPageObjectProvider(
-			new InfoItemReference(
-				FileEntry.class.getName(), Long.valueOf(urlTitle)));
+		FriendlyURLEntry friendlyURLEntry =
+			_friendlyURLEntryLocalService.fetchFriendlyURLEntry(
+				groupId, FileEntry.class, urlTitle);
+
+		if (friendlyURLEntry != null) {
+			return getLayoutDisplayPageObjectProvider(
+				new InfoItemReference(
+					FileEntry.class.getName(), friendlyURLEntry.getClassPK()));
+		}
+
+		if (Validator.isNumber(urlTitle)) {
+			return getLayoutDisplayPageObjectProvider(
+				new InfoItemReference(
+					FileEntry.class.getName(), GetterUtil.getLong(urlTitle)));
```

Comment:
I have a question, more ti know why and  get it clearer.

Why did you decided to use the getterUtil over the Long.valueOf here?